### ISSUE TITLE
Inconsistent SSH Behavior between Tasks and Shells in VS Code Desktop

### ISSUE THREAD

**Issue by burningion** [gitpod-io/gitpod#16440]
### Bug description

When you open up VS Code Desktop, Tasks do not have the environment variable `SSH_ AUTH_SOCK` set properly. This means you cannot do SSH Agent forwarding within Tasks, and cannot sign commits using SSH keys.

Doing an `env | grep SSH` on a fresh task, you get nothing.

Doing an `env | grep SSH` on a created bash shell, you get a socket that works, and works with SSH Agent forwarding.

If I try to manually spin up the ssh agent in a Task shell, I get:

```bash
$ env | grep SSH
$
```

This shows that there is no `SSH_AUTH_SOCK` set. I can try to create it using `ssh-agent`:

```bash
$ eval $(ssh-agent -s)
Agent pid 4250
$ env | grep SSH
SSH_AUTH_SOCK=/tmp/ssh-xPzWFGXpsd6T/agent.4249
SSH_AGENT_PID=4250
```

But if I try to verify my ssh agent forwarding with Github (I have an ssh key on my local machine and ssh agent forwarding enabled):

```bash
$ ssh -T git@github.com
ssh -T git@github.com
git@github.com: Permission denied (publickey).
```
I still do not get SSH agent forwarding. Once I set the `SSH_AUTH_SOCK` to one created by a Shell terminal (created and assigned automatically within every shell created with the plus sign in VS Code Desktop), I get authentication and signing with SSH agent forwarding both working properly.

```bash
# in the new bash shell
$ env | grep SSH
SSH_AUTH_SOCK=/tmp/vscode-ssh-auth-sock-494130870
SSH_CONNECTION=UNKNOWN 65535 UNKNOWN 65535
SSH_CLIENT=UNKNOWN 65535 65535

# exported in the Task
$ export SSH_AUTH_SOCK=/tmp/vscode-ssh-auth-sock-494130870
$ ssh -T git@github.com
Hi burningion! You've successfully authenticated, but GitHub does not provide shell access.
```

### Steps to reproduce

See the above

### Expected behavior

I expect consistent behavior across Tasks and newly created Shells for ssh agent forwarding. Otherwise I can't do Git commit signing with SSH keys from within the native UI, and have to manually open a shell and type out my commits.

**Comment by burningion** [comment]
I think this would actually be fixed by https://github.com/gitpod-io/gitpod/issues/16108

**Comment by jeanp413** [comment]
Slack discussion : https://gitpod.slack.com/archives/C020VCB0U5A/p1676470768617929

TL;DR: task terminal does not have agent forwarding because they are created before ssh connection happens

**Comment by akosyakov** [comment]
I guess we could make supervisor to install own SSH_AUTH_SOCK and proxy it to IDE via an extension. Kind of the same what we do for gp open/preview.
